.github/workflows/gradle-test.pr.yml

```diff
@@ -9,12 +9,14 @@ jobs:
         version: [ 8, 11, 17 ]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-java@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
         with:
           distribution: 'zulu'
           java-version: ${{ matrix.version }}
-      - run: ./gradlew clean test --stacktrace
+      - uses: gradle/gradle-build-action@v2
+        with:
+          arguments: clean test --stacktrace
```

Comment:
https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.